core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -79,11 +81,24 @@ public static void registerAll(ObjectMapper mapper) {
         .addDeserializer(UpdateRequirement.class, new UpdateRequirementDeserializer())
         .addSerializer(OAuthTokenResponse.class, new OAuthTokenResponseSerializer())
         .addDeserializer(OAuthTokenResponse.class, new OAuthTokenResponseDeserializer())
-        .addSerializer(ReportMetricsRequest.class, new ReportMetricsRequestSerializer<>())
-        .addDeserializer(ReportMetricsRequest.class, new ReportMetricsRequestDeserializer<>())
-        .addSerializer(ImmutableReportMetricsRequest.class, new ReportMetricsRequestSerializer<>())
-        .addDeserializer(
-            ImmutableReportMetricsRequest.class, new ReportMetricsRequestDeserializer<>());
+        .addSerializer(ReportMetricsRequest.class, new ReportMetricsRequestSerializer())
+        .addDeserializer(ReportMetricsRequest.class, new ReportMetricsRequestDeserializer())
+        .setDeserializerModifier(
```

Comment:
Do we want to do this? It seems cleaner just to add all of the concrete classes explicitly.

Reply:
I'm 50/50 on this and essentially both ways work. With the current version of `master` the only downside is that a deserializer would have to be made generic in order to reuse it. Other than that. The downside with this PR is that the solution is much longer than I'd like it to be

Reply:
I'd lean toward not doing this then. It's easy to understand what is happening with basic registration, especially compared to this where we're introducing a new feature with a callback. And the type parameter on the deserializer isn't bad.

Reply:
works for me, closing this PR then